installer/bundle/bundle_skel_AIX.sh

```diff
@@ -0,0 +1,432 @@
+            exit 0
+            ;;
+
+        --debug)
+            echo "Starting shell debug mode." >&2
+            echo "" >&2
+            echo "SCRIPT_INDIRECT: $SCRIPT_INDIRECT" >&2
+            echo "SCRIPT_DIR:      $SCRIPT_DIR" >&2
+            echo "EXTRACT DIR:     $EXTRACT_DIR" >&2
+            echo "SCRIPT:          $SCRIPT" >&2
+            echo >&2
+            set -x
+            shift 1
+            ;;
+
+        -? | --help)
+            usage `basename $0` >&2
+            cleanup_and_exit 0
+            ;;
+
+        *)
+            usage `basename $0` >&2
+            cleanup_and_exit 1
+            ;;
+    esac
+done
+
+if [ -z "${installMode}" ]; then
+    echo "$0: No options specified, specify --help for help" >&2
+    cleanup_and_exit 3
+fi
+
+#
+# Note: From this point, we're in a temporary directory. This aids in cleanup
+# from bundled packages in our package (we just remove the diretory when done).
+#
+
+mkdir -p $EXTRACT_DIR
+cd $EXTRACT_DIR
+
+# Do we need to remove the package?
+if [ "$installMode" = "R" -o "$installMode" = "P" ]
+then
+    if [ -f /opt/microsoft/scx/bin/uninstall ]; then
+        /opt/microsoft/scx/bin/uninstall $installMode
+    else
+        # This is an old kit.  Let's remove each separate provider package
+        for i in /opt/microsoft/*-cimprov; do
+            PKG_NAME=`basename $i`
+            if [ "$PKG_NAME" != "*-cimprov" ]; then
+                echo "Removing ${PKG_NAME} ..."
+                pkg_rm ${PKG_NAME}
+            fi
+        done
+
+        # Now just simply pkg_rm scx (and omi if it has no dependencies)
+        pkg_rm scx
+        pkg_rm omi
+    fi
+
+    if [ "$installMode" = "P" ]
+    then
+        echo "Purging all files in cross-platform agent ..."
+        rm -rf /etc/opt/microsoft/*-cimprov /etc/opt/microsoft/scx /opt/microsoft/*-cimprov /opt/microsoft/scx /var/opt/microsoft/*-cimprov /var/opt/microsoft/scx
+        rmdir /etc/opt/microsoft /opt/microsoft /var/opt/microsoft 1>/dev/null 2>/dev/null
+
+        # If OMI is not installed, purge its directories as well.
+        check_if_pkg_is_installed omi
+        if [ $? -ne 0 ]; then
+            rm -rf /etc/opt/omi /opt/omi /var/opt/omi
+        fi
+    fi
+fi
+
+if [ -n "${shouldexit}" ]
+then
+    # when extracting script/tarball don't also install
+    cleanup_and_exit 0
+fi
+
+#
+# Do stuff before extracting the binary here, for example test [ `id -u` -eq 0 ],
+# validate space, platform, uninstall a previous version, backup config data, etc...
+#
+
+#
+# Extract the binary here.
+#
+
+echo "Extracting..."
+
+tail -n +${SCRIPT_LEN_PLUS_ONE} "${SCRIPT}" | gunzip -c | tar xf -
+STATUS=$?
+if [ ${STATUS} -ne 0 ]
+then
+    echo "Failed: could not extract the install bundle."
+    cleanup_and_exit ${STATUS}
+fi
+
+#
+# Do stuff after extracting the binary here, such as actually installing the package.
+#
+
+EXIT_STATUS=0
+SCX_EXIT_STATUS=0
+OMI_EXIT_STATUS=0
+BUNDLE_EXIT_STATUS=0
```

Comment:
I imagine, since bundles don't exist on UNIX, that BUNDLE_EXIT_STATUS won't exist either when you clean up the code below.